### PR TITLE
test: update k8s to 1.8.14, 1.10.4 and 1.11.0

### PR DESCRIPTION
**Pull request by aanm**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4634)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
Open again because the CI was not triggering https://github.com/cilium/cilium/pull/4538

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: ginkgo-kubernetes-all.Jenkinsfile [hunk 1]
```diff
@@ -84,6 +85,9 @@ pipeline {
                     "K8s-1.9":{
                         sh 'cd ${TESTDIR}; K8S_VERSION=1.9 ginkgo --focus=" K8s*" -v --failFast=${FAILFAST}'
                     },
+                    "K8s-1.10":{
```
Comment (eloycoto) [inline ginkgo-kubernetes-all.Jenkinsfile]: I know that it is still wip, but running 3 different version of Kubernetes in the same stage is not a good idea IMHO.  I would use two k8s versions per stage.

Comment (aanm) [reply ginkgo-kubernetes-all.Jenkinsfile]: Should I create a new stage? :thinking:

Comment (eloycoto) [reply ginkgo-kubernetes-all.Jenkinsfile]: I would add that one in the non-release ones, and change the names, mora than two are too much for the host IMHO

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
```
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-K8s/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:376
Cannot curl from "testclient-6bq72" to testds-service
Expected command: kubectl exec -n default testclient-6bq72 -- curl -s -D /dev/stderr --fail --connect-timeout 5 --max-time 5 http://testds-service.default.svc.cluster.local:80/ 
To succeed, but it failed:
Exitcode: 28 
Stdout:
 	 
Stderr:
 	 command terminated with exit code 28
```
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/444/

**Comment by aanm** [comment]
test-missed-k8s

**Review by eloycoto — APPROVED** [review]

**Comment by eloycoto** [comment]
Not block, but maybe we need to change the following functions: 
```
kubectl.WaitKubeDNS()
kubectl.WaitForKubeDNSEntry()
```

I don't think that it's block, but as soon as 1.11 get into master maybe the builds are getting fragile because we're not waiting for the DNS correctly.

**Comment by tgraf** [comment]
> I don't think that it's block, but as soon as 1.11 get into master maybe the builds are getting fragile because we're not waiting for the DNS correctly.

@eloycoto Can you elaborate? What does getting into master mean in this case?

**Comment by eloycoto** [comment]
If we test by default in master using Kubernetes 1.11, we're going to switch from Kube-dns to Core-DNS and we don't have specific code to wait until core-dns pods are ready, so maybe we hit some flakes due pods are not ready.

**Comment by eloycoto** [comment]
For the track:

```

vagrant@k8s1:~$ kubectl get nodes
NAME      STATUS     ROLES     AGE       VERSION
k8s1      NotReady   master    2h        v1.11.0-beta.0
k8s2      NotReady   <none>    2h        v1.11.0-beta.0
vagrant@k8s1:~$ kubectl -n kube-system get pods -l k8s-app=kube-dns
NAME                       READY     STATUS    RESTARTS   AGE
coredns-7997f8864c-5w7fk   0/1       Pending   0          2h
coredns-7997f8864c-zdnhn   0/1       Pending   0          2h
vagrant@k8s1:~$ kubectl -n kube-system get services
NAME       TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)         AGE
kube-dns   ClusterIP   10.96.0.10   <none>        53/UDP,53/TCP   2h
vagrant@k8s1:~$
```

No changes are needed at all.

**Comment by eloycoto** [comment]
This PR was reverted. Not backporting to 1.1

**Pull request merged**